### PR TITLE
Automated cherry pick of #86346 #87609 #88684 #89319 #89444 upstream release 1.17

### PR DESCRIPTION
**Pull request by saad-ali**
**What type of PR is this?**
/kind bug
/sig storage
/priority important-soon

**What this PR does / why we need it**:
Cherry pick PRs #86346 #87609 #88684 #89319 #89444 from master to upstream release 1.17
/assign @liggitt 
CC @SataQiu @andyzhangx @jsafrane @mboersma @gnufied 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
AzureFile and CephFS use new Mount library that prevents logging of sensitive mount options.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by andyzhangx** [comment]
/hold cancel
hold for wrong PR...

**Review by andyzhangx — APPROVED** [review]
LGTM on windows and azure part

**Comment by saad-ali** [comment]
Thanks @andyzhangx

**Comment by liggitt** [comment]
This is a much larger change than I was expecting (and touches several areas we saw last-minute regressions/fixes in in the 1.18 dev cycle). @saad-ali, can you characterize the regression risks and test coverage for the areas this touched to help weigh the risk of this backport (xref https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md#what-kind-of-prs-are-good-for-cherry-picks)?

File: vendor/k8s.io/utils/mount/mount_helper_common.go [hunk 1]
```diff
@@ -73,9 +73,13 @@ func doCleanupMountPoint(mountPath string, mounter Interface, extensiveMountPoin
 		return err
 	}
 
-	notMnt, mntErr := mounter.IsLikelyNotMountPoint(mountPath)
-	if mntErr != nil {
-		return mntErr
+	if extensiveMountPointCheck {
```
Comment (kvaps) [inline vendor/k8s.io/utils/mount/mount_helper_common.go:76]: ref https://github.com/kubernetes/kubernetes/issues/89181

**Comment by saad-ali** [comment]
> This is a much larger change than I was expecting (and touches several areas we saw last-minute regressions/fixes in in the 1.18 dev cycle). @saad-ali, can you characterize the regression risks and test coverage for the areas this touched to help weigh the risk of this backport (xref https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md#what-kind-of-prs-are-good-for-cherry-picks)?

Agreed this is a very large change, and given the way the mount library is structured may pick up other unintended changes. There are lots of storage tests that exercice the mount library, but they are only run against a subset of possible volume plugins. I would be ok holding off on the cherry pick for these changes.

**Comment by tpepper** [comment]
/hold
per Saad

**Pull request closed without merging**